src/ape_cache/query.py

```diff
@@ -236,12 +237,15 @@ def perform_query(self, query: QueryType) -> Optional[List]:  # type: ignore
                     # NOTE: Should be unreachable if estimated correctly
                     raise QueryEngineError(f"Could not perform query:\n{query}")
 
-                breakpoint()
                 for row in result:
-                    yield {key: value for (key, value) in row.items()}
-                # return [QueryType({key: value for (key, value) in row.items()} for row in result)]
-                # for row in result:
-                #     yield {key: value for (key, value) in row.items()}
+                    row_dict = {key: value for (key, value) in row.items()}
+                    row_dict["hash"] = HexBytes(row_dict["hash"])
+                    row_dict["parentHash"] = HexBytes(row_dict.pop("parent_hash"))
+                    row_dict["gasLimit"] = row_dict.pop("gas_limit")
+                    row_dict["gasUsed"] = row_dict.pop("gas_used")
+                    row_dict["baseFeePerGas"] = row_dict.pop("base_fee")
+                    row_dict["totalDifficulty"] = row_dict.pop("total_difficulty")
+                    yield Block(**row_dict)
```

Comment:
I think it is cool to only support EVM blocks for now, but it may be misleading in the CLI where you specify an ecosystem name and then it uses `Block` class from `ape_ethereum`

Reply:
Somehow I think it is still working on Starknet though haha!

Reply:
I only did this to make it work, we can change this however we want. I have to add `singledispatchmethod` to this method so we can handle transactions and events.

Reply:
Yeah, this was prototyping a work in progress, probably this method from `EcosystemAPI` is better:
https://github.com/ApeWorX/ape/blob/main/src/ape/api/networks.py#L142

Reply:
@fubuloubu if we decode block, it causes conflicts with the column names of the table and it causes conflicts when we don't query all columns